src/utils/job_params_util.py

```diff
@@ -61,3 +79,276 @@ def get_resource_params_from_job_params(params):
         "gpu": gpu.to_dict(),
         "gpu_memory": gpu_memory.to_dict(),
     }
+
+
+class JobParams(object):
+    subclasses = {}
+
+    @classmethod
+    def register_subclass(cls, job_type):
```

Comment:
It looks strange to me, since we do not need to be able to register subclass at runtime, why we need this method instead of just put a `subclasses = {"RegularJob": RegularJobParams ...}`?